presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        transaction = icebergTable.newTransaction();
+
+        // delete before insert .. simulating overwrite
+        executeDelete(session, tableHandle);
+
+        // Grab the snapshot-ids of the source table. The table scans for insert are planned and
+        // executed as of these snapshots and these snapshot ids are written to storage table
+        // as 'dependsOnTables' property. This 'as-of' processing ensures that the metadata of the
+        // materialized view and the data are in sync.
+        Map<ConnectorTableHandle, Long> tableSnapshotIdMap = new HashMap<>();
+        for (ConnectorTableHandle handle : sourceTableHandles) {
+            Optional<io.prestosql.plugin.iceberg.TableToken> token = getTableToken(session, handle);
+            if (token.isPresent()) {
+                tableSnapshotIdMap.put(handle, token.get().getSnapshotId());
+            }
+            else {
+                tableSnapshotIdMap.put(handle, (long) -1);
+            }
+        }
+        transactionHandle.setTableSnapshotIdMap(tableSnapshotIdMap);
+        return new IcebergWritableTableHandle(
+            table.getSchemaName(),
+            table.getTableName(),
+            SchemaParser.toJson(icebergTable.schema()),
+            PartitionSpecParser.toJson(icebergTable.spec()),
+            getColumns(icebergTable.schema(), typeManager),
+            getDataPath(icebergTable.location()),
+            getFileFormat(icebergTable));
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session,
+            ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments,
+            Collection<ComputedStatistics> computedStatistics,
+            List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
+        org.apache.iceberg.Table icebergTable = transaction.table();
+        List<CommitTaskData> commitTasks = new ArrayList<>();
+
+        if (!fragments.isEmpty()) {
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(HadoopInputFile.fromLocation(task.getPath(), getConfiguration(session, table.getSchemaName())))
```

Comment:
Use `HdfsInputFile` instead of `HadoopInputFile`. See `finishInsert()`

Reply:
Got it. done.